static/specification.html

```diff
@@ -639,30 +635,50 @@ <h2>
         [SecureContext, Exposed=Window]
         interface MonetizationEvent : Event {
           constructor(DOMString type, MonetizationEventInit eventInitDict);
-          readonly attribute DOMString amount;
-          readonly attribute DOMString assetCode;
-          readonly attribute octet assetScale;
+          readonly attribute DOMString? amount;
+          readonly attribute DOMString? assetCode;
+          readonly attribute octet? assetScale;
           readonly attribute DOMString? receipt;
+          readonly attribute PaymentCurrencyAmount amountSent;
```

Comment:
Note that the payments `PaymentCurrencyAmount` is a dictionary type, not interface, and thus not suitable for an event attribute. If we were to go down this route, we would have to define a new interface.

```
[SecureContext, Exposed=Window]
interface MonetizationAmount {
  readonly attribute DOMString value;
  readonly attribute DOMString currency;
};

[SecureContext, Exposed=Window]
interface MonetizationEvent : Event {
  constructor(DOMString type, MonetizationEventInit eventInitDict);
  readonly attribute MonetizationAmount amountSent;
```

While this would mean window.MonetizationAmount would be exposed, the constructor wouldn’t actually be. @sidvishnoi also says this is standard practice. There used to be a [NoInterfaceObject] decoration, but it was removed, and people just prefix the interfaces.

Reply:
@marcoscaceres - you were close to the design of `PaymentCurrencyAmount `. WDYT?

Should we define a `MonetizationAmount` interface that aligns with the types of the `PaymentCurrencyAmount` dictionary?

Reply:
@sublimator - what if we simply have the following:

```
[SecureContext, Exposed=Window]
interface MonetizationEvent : Event {
  constructor(DOMString type, MonetizationEventInit eventInitDict);
  readonly attribute DOMString amountSent;
  readonly attribute DOMString amountSentCurrency;
```

At least these data types align with the `PaymentCurrencyAmount`

Reply:
and amountSentAssetCode I guess ?

Reply:
> There used to be a [NoInterfaceObject] decoration, but it was removed, and people just prefix the interfaces.

Sadly missed 

> what if we simply have the following

In practice that might not be too bad:
```
const {amountSent, amountSentCurrency} = event; 
```

Reply:
This feels marginally nicer...
` const { amountSent : { value, currency }} = event;`

Reply:
In 2022, using a compound type as an attribute on an event shouldn't really be this hard of a decision eh? WebIDL should be fixed. 

> This feels marginally nicer...
> const { amountSent : { value, currency }} = event;

You know you want it! Declare it so

Reply:
Done